src/main/java/net/minecraftforge/common/ToolModificationResult.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common;
+
+import com.google.common.base.Objects;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+/**
+ * Holds a {@link #toolModifiedState() tool-modified state} and whether the modification {@link #passed()} or {@link #failed()}.
+ * <p>
+ * If the tool modification result has a fail state, the tool-modified state is always an empty optional.
+ * <p>
+ * If the tool modification result has a pass state, an empty tool-modified state signifies a fallback on the default behavior.
+ */
+public final class ToolModificationResult
+{
+    private static final ToolModificationResult FAIL_RESULT = new ToolModificationResult(Type.FAIL, null);
+    private static final ToolModificationResult EMPTY_PASS_RESULT = new ToolModificationResult(Type.PASS, null);
+    private final Type type;
+    @Nullable
+    private final BlockState toolModifiedState;
+
+    private ToolModificationResult(Type type, @Nullable BlockState toolModifiedState)
+    {
+        this.type = type;
+        this.toolModifiedState = toolModifiedState;
+    }
+
+    /**
+     * Signifies that this action failed and should be canceled immediately.
+     *
+     * @return An immutable tool modification result that signifies a fail state
+     */
+    @NotNull
+    public static ToolModificationResult fail()
+    {
+        return FAIL_RESULT;
+    }
+
+    /**
+     * Signifies that the default behavior should be invoked.
+     *
+     * @return An immutable tool modification result that signifies a fallback on default behavior
+     */
+    @NotNull
+    public static ToolModificationResult passEmpty()
```

Comment:
`passWithDefaultBehaviour` better describes the usecase of the method, but this is nitpicking at this point.

Reply:
I don't like "With", maybe `passFallbackBehavior` or simply `passFallback`?

Reply:
The problem is that you are indicating two different things.
1) The fact that you want the result to indicate a `pass` as primary behavior
and
2) The fact that the default behavior should be executed

`passFallback` or `passFallbackBehaviour` don't indicate that the results indicates _both_ of these behaviors.

But I would already agree with `passToFallbackBehaviour` or something in that direction.

Reply:
Got it.